### ISSUE TITLE
Disabling the es6 env doesn't work

### ISSUE THREAD

**Issue by mzgol** [eslint/eslint#4666]
I'm using [eslint-config-mzgol](https://www.npmjs.com/package/eslint-config-mzgol) with a config in the root of the project directory:
```json
{
    "extends": "mzgol",

    "root": true,

    "env": {
        "node": true
    },

    "rules": {
        "spaced-comment": 0,
        "prefer-spread": 0,
        "prefer-reflect": 0
    }
}
```
Then, in the directory `public/js` I have another `.eslintrc`:
```json
{
    "env": {
        "es6": false,
        "browser": true,
        "node": false
    },

    "rules": {
        "no-var": 0,
        "object-shorthand": 0,
        "prefer-arrow-callback": 0,
        "prefer-const": 0,
        "prefer-spread": 0,
        "prefer-reflect": 0,
        "prefer-template": 0
    }
}
```

and yet ESLint doesn't fail on ES6 code inside of this directory.

ESLint 1.10.3.

**Comment by eslintbot** [comment]
@mzgol Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
Can you provide some sample code that you are expecting to trigger a warning?

**Comment by mzgol** [comment]
@nzakas E.g.:
```js
var data = location.search
    .replace(/^\?/, '')
    .replace(/&+$/, '')
    .split('&')
    .map(pairString => pairString.split('='))
```

I see the error if I go to `node_modules/eslint-config-mzgol/eslint-config-mzgol.js` and comment out the `es6: true` part.

**Comment by mysticatea** [comment]
I'm sorry, this is duplicate of #3915 (PR was #3958 but )

**Comment by nzakas** [comment]
Ooh right, let's track in #3915.

**Comment by mzgol** [comment]
Thanks, I wouldn't find it by the title of #3915. :-)
